configure

```diff
@@ -2449,6 +2449,9 @@ sub write_cmake_file {
       $var =~ s/-/_/g;
       my $val = $opts_{$k};
 
+      if($var eq "OPENDDS_COMPILER" && $val eq "cl") {
+        $val = "\${CMAKE_CXX_COMPILER}";
+      }
```

Comment:
`OPENDDS_COMPILER` isn't being used except in the tests, where it's iffy if it should be used.

Reply:
If the feature being added by this PR can work without a special case here, that would be preferred.

Reply:
The reason for this change is the tests are replacing CMAKE_CXX_COMPILER with it.  

https://github.com/objectcomputing/OpenDDS/blob/67ec6d2445a0925724fb84935729e5789b4e502a/tests/cmake_integration/Messenger/Messenger_2/CMakeLists.txt#L12

This causes problems with the main part of this change which is in the macro where we change the OpenDDS and Tao options.  

If we want to remove the CMAKE_CXX_COMPILER replacement with OPENDDS_COMPILER from our tests, then we can get rid of the configure change.

Reply:
> This causes problems with the main part of this change which is in the macro where we change the OpenDDS and Tao options.

What are the problems?

> If we want to remove the CMAKE_CXX_COMPILER replacement with OPENDDS_COMPILER from our tests, then we can get rid of the configure change.

I don't really like overriding `CMAKE_CXX_COMPILER`, but looking at the original commit (117f95dcd0ca87ad8aebaa3a92100874a5c2a570 in #1240), it looks like I was trying to account for if we built OpenDDS with a compiler that's not the system compiler or what ever CMake defaults to. So we could remove them, but we'd have to make sure we're always setting the compiler if we need to.

Reply:
CMake's documentation of `CMAKE_<LANG>_COMPILER` says we should not use it this way (in `cmake-variables(7)`).

When running the scripts for `tests/cmake_integration`, the `cmake` command line should contain `-DCMAKE_CXX_COMPILER=foo` if that build doesn't use the system's default compiler.

If you know it's using gnuace you could run make to determine how ACE was configured:
```
 make -C $ACE_ROOT/ace -npf GNUmakefile.ACE | grep '^CXX ='
```

OpenDDS's CMake Module could (if we wanted to implement this) attempt to detect a mismatch and at least warn.

Reply:
> 
> What are the problems?
> 

The line

list(APPEND _opendds_options "-Yp,${CMAKE_CXX_COMPILER}")

would be putting something like "-Yp,cl" into the options, rather than "-Yp,<Path To cl.exe>"

Reply:
> When running the scripts for tests/cmake_integration, the cmake command line should contain -DCMAKE_CXX_COMPILER=foo if that build doesn't use the system's default compiler.

Yes, that's what I was referring to with "setting the compiler if we need to".

> OpenDDS's CMake Module could (if we wanted to implement this) attempt to detect a mismatch and at least warn.

This might be possible, but we'd have to do some processing on both of the variables to resolve them to their canonical paths. Also [`CMAKE_CXX_COMPILER` can contain arguments](https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_COMPILER.html) which makes comparing a little harder.

> would be putting something like "-Yp,cl" into the options, rather than "-Yp,<Path To cl.exe>"

On Linux I think `tao_idl` only gets `g++` by default, so I would think letting it use the `PATH` on Windows would be okay for the tests. If you're worried about it messing with the goal of the PR, it shouldn't, because users shouldn't be using `set(CMAKE_CXX_COMPILER ${OPENDDS_COMPILER})`. Only our tests should be doing that; [DevGuide Messenger, the "real" CMake example, doesn't have it](https://github.com/objectcomputing/OpenDDS/blob/master/DevGuideExamples/DCPS/Messenger/CMakeLists.txt). The tests should have compiler on the `PATH`, otherwise default `CMAKE_CXX_COMPILER` should be an absolute path, at least as far as I know.